.github/workflows/terraform-ci.yml

```diff
@@ -0,0 +1,218 @@
+      with:
+        go-version: ${{ needs.get-go-version.outputs.go-version }}
+    - name: Install base apps
+      run: |
+        sudo apt-get install -y expect openssl jq
+    - name: Install gotestsum
+      run: |
+        curl -sSL "https://github.com/gotestyourself/gotestsum/releases/download/v${{ env.GOTESTSUM_VERSION }}/gotestsum_${{ env.GOTESTSUM_VERSION }}_linux_amd64.tar.gz" | \
+        tar -xz --overwrite -C /usr/local/bin gotestsum
+    - name: Install AWS CLI
+      run: |
+        curl "https://s3.amazonaws.com/session-manager-downloads/plugin/latest/ubuntu_64bit/session-manager-plugin.deb" -o "session-manager-plugin.deb"
+        sudo dpkg -i session-manager-plugin.deb
+        aws --version
+        echo session-manager-plugin version "$(session-manager-plugin --version)"
+    - name: Install AWS ECS CLI
+      run: |
+        curl -sSL "https://amazon-ecs-cli.s3.amazonaws.com/ecs-cli-linux-amd64-latest" -o /usr/local/bin/ecs-cli
+        chmod +x /usr/local/bin/ecs-cli
+        ecs-cli --version
+    - name: Assume AWS IAM Role
+      uses: aws-actions/configure-aws-credentials@e1e17a757e536f70e52b5a12b2e8d1d1c60e04ef # v2.0.0
+      with:
+        role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+        aws-region: ${{ secrets.AWS_REGION }}
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        role-duration-seconds: 3600
+    - name: terraform init & apply
+      run: |
+        cd setup-terraform/
+        VARS="-var tags={\"build_url\":\"$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID\"}"
+        VARS+=' -var launch_type=${{ matrix.launch-type }}'
+        VARS+=' -var consul_version=${{ matrix.consul-version }}'
+        case $GITHUB_REF_NAME in
+            main | release/*) VARS+=" -var enable_hcp=${{ matrix.enable-hcp }}";;
+            *) VARS+=" -var enable_hcp=false";;
+        esac
+        terraform init
+        terraform apply -auto-approve $VARS
+    - name: Acceptance tests
+      run: |
+        mkdir -p "$TEST_RESULTS"
+        cd tests/
+        gotestsum --junitfile "$TEST_RESULTS/gotestsum-report.xml" --format standard-verbose -- ./... -p 1 -timeout 1h -v -failfast
+    - uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+      if: always()
+      with:
+        name: acceptance-test-results
+        path: ${{ env.TEST_RESULTS }}/gotestsum-report.xml
+    - name: terraform destroy
+      if: always()
+      run: |
+        cd setup-terraform/
+        VARS="-var tags={\"build_url\":\"$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID\"}"
+        VARS+=' -var launch_type=${{ matrix.launch-type }}'
+        VARS+=' -var consul_version=${{ matrix.consul-version }}'
+        case $GITHUB_REF_NAME in
+            main | release/*) VARS+=" -var enable_hcp=${{ matrix.enable-hcp }}";;
+            *) VARS+=" -var enable_hcp=false";;
+        esac
```

Comment:
There might be a cleaner way to accomplish this in Github actions than there was in Circle CI.

Maybe:

* Create a workflow the runs for release branches and main
* Create another workflow for pull requests / other branches

Both of those workflows can trigger a [reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows) that passes in an `enable_hcp=true/false` setting and runs acceptance tests.

(Probably not worth the effort)

Reply:
This is a good idea and I did give it some thought. I think we should punt this down the road for now because I'm supposed to have this all wrapped up by Friday (Mar 24th.)